bugmon_tc/monitor/tasks.py

```diff
@@ -103,41 +65,136 @@ def task(self):
+        :param bug_id: Bug ID
+        :param src: Path to monitor artifact
+        :param kwargs: Additional options
+        """
+        super().__init__(parent_id, bug_id)
+        self.id = slugId()
+        self.parent_id = parent_id
+        self.bug_id = bug_id
+        self.monitor_path = monitor_path
+        self.dest = f"processor-result-{self.bug_id}-{self.parent_id}.json"
+
+        self.trace_dest = None
+        if use_pernosco:
+            self.trace_dest = f"processor-rr-trace-{bug_id}-{parent_id}.tar.gz"
+
+        self.force_confirm = force_confirm
 
     @property
     def env(self):
         """Environment variables for the task"""
-        return {
+        env_object = {
+            "BUG_ACTION": "process",
+            "MONITOR_ARTIFACT": str(self.monitor_path),
+            "PROCESSOR_ARTIFACT": str(self.dest),
+        }
+
+        if self.force_confirm:
+            env_object["FORCE_CONFIRM"] = "1"
+
+        if self.trace_dest:
+            env_object["TRACE_ARTIFACT"] = str(self.trace_dest)
+
+        return env_object
+
+    @property
+    def scopes(self):
+        """Scopes applied to the task"""
+        scopes = super().scopes
+        scopes.append(f"queue:get-artifact:project/fuzzing/bugmon/{self.monitor_path}")
+
+        return scopes
+
+    @property
+    def worker_type(self):
+        """The worker type to use for this task"""
+        # If a trace path was supplied, use the bugmon-pernosco worker
+        if self.trace_dest:
+            return "bugmon-pernosco"
+
+        return "bugmon-processor"
+
+
+class ReporterTask(BaseTask):
+    """Helper class for generating reporter tasks"""
+
+    def __init__(self, parent_id, bug_id, process_path, dep, trace_path=None):
+        """Instantiate a new ReporterTask instance.
+
+        :param parent_id: ID of parent task
+        :param bug_id: Bug ID
+        :param process_path: Path to process artifact
+        :param dep: Task dependency
+        :param trace_path: Optional path to trace artifact.
+        """
+        super().__init__(parent_id, bug_id)
+        self.process_path = process_path
+
+        self.dependency = dep
+        self.trace_dest = trace_path
+
+    @property
+    def env(self):
+        """Environment variables for the task"""
+        env_object = {
             "BUG_ACTION": "report",
-            "PROCESSOR_ARTIFACT": self.src,
+            "PROCESSOR_ARTIFACT": str(self.process_path),
         }
 
+        if self.trace_dest is not None:
+            env_object["TRACE_ARTIFACT"] = str(self.trace_dest)
+
+        return env_object
+
     @property
     def scopes(self):
         """Scopes applied to the task"""
-        return [
-            "docker-worker:capability:device:hostSharedMemory",
-            "docker-worker:capability:device:loopbackAudio",
-            "queue:scheduler-id:-",
-            f"queue:get-artifact:project/fuzzing/bugmon/{self.src}",
-            "secrets:get:project/fuzzing/bz-api-key",
```

Comment:
Where do the bz and pernosco credentials come from? If you're getting them in the docker script and setting them in the environment, you'll still need the scopes to access them if they're stored as secrets.

Reply:
Thanks.  I had to add them and the bz-api-key back.